From 57ba1db30555d90a909e20c4752654a646922bdc Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Fri, 30 Aug 2013 15:29:10 +0800
Subject: [PATCH 0568/5242] MLK-11703-3: power: sabresd_battery: remove check
 charger offset on mx6slevk

commit  fb0baf0f41d94b3cb710b58d567687d06fa9d8dc from
https://source.codeaurora.org/external/imx/linux-imx.git

cherry-pick below patch:

ENGR00277663-3 power: sabresd_battery: remove check charger offset on mx6slevk

No need check the charger offset on mx6slevk, since there is no adc converter

Signed-off-by: Robin Gong <b38343@freescale.com>
(cherry picked from commit 7683c62cf2290629f09983744d5a3987a4d64669)
(cherry picked from commit 5f29715585ae82a435aa5ba0fe6393a53c10f87d)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/power/supply/sabresd_battery.c |   25 ++++++++++++++-----------
 1 file changed, 14 insertions(+), 11 deletions(-)

diff --git a/drivers/power/supply/sabresd_battery.c b/drivers/power/supply/sabresd_battery.c
index c8c07b3..2181a6f 100644
--- a/drivers/power/supply/sabresd_battery.c
+++ b/drivers/power/supply/sabresd_battery.c
@@ -622,17 +622,20 @@ static struct max8903_pdata *max8903_of_populate_pdata(
 		return NULL;
 	}
 
-	if (of_property_read_u32(of_node, "offset-charger",
-		&offset_charger))
-		dev_err(dev, "Not setting offset-charger in dts!\n");
-
-	if (of_property_read_u32(of_node, "offset-discharger",
-		&offset_discharger))
-		dev_err(dev, "Not setting offset-discharger in dts!\n");
-
-	if (of_property_read_u32(of_node, "offset-usb-charger",
-		&offset_usb_charger))
-		dev_err(dev, "Not setting offset-usb-charger in dts!\n");
+	/* no need check offset without adc converter */
+	if (!pdata->feature_flag) {
+		if (of_property_read_u32(of_node, "offset-charger",
+					 &offset_charger))
+			dev_err(dev, "Not setting offset-charger in dts!\n");
+
+		if (of_property_read_u32(of_node, "offset-discharger",
+					 &offset_discharger))
+			dev_err(dev, "Not setting offset-discharger in dts!\n");
+
+		if (of_property_read_u32(of_node, "offset-usb-charger",
+					 &offset_usb_charger))
+			dev_err(dev, "Not setting offset-usb-charger in dts!\n");
+	}
 
 	return pdata;
 }
-- 
1.7.9.5

